PerimeterXModule/Internals/Cookies/PxCookieUtils.cs

```diff
@@ -1,29 +1,28 @@
 ﻿using Jil;
 using PerimeterX.DataContracts.Cookies;
 using System;
+using System.Collections.Generic;
 using System.Security.Cryptography;
 using System.Text;
 
 namespace PerimeterX
 {
 	public static class PxCookieUtils
 	{
-		public static IPxCookie BuildCookie(PxModuleConfigurationSection config, PxContext context, ICookieDecoder cookieDecoder)
+		public static IPxCookie BuildCookie(PxModuleConfigurationSection config, Dictionary<string,string> cookies, ICookieDecoder cookieDecoder)
 		{
-			if (context.PxCookies.Count > 0)
+			if (cookies.Count == 0)
 			{
-				var cookie = context.PxCookies.ContainsKey(PxConstants.COOKIE_V3_PREFIX) ? context.PxCookies[PxConstants.COOKIE_V3_PREFIX] : null;
-				if (cookie != null)
-				{
-					return new PxCookieV3(cookieDecoder, cookie);
-				}
-				cookie = context.PxCookies[PxConstants.COOKIE_V1_PREFIX];
-				if (cookie != null)
-				{
-					return new PxCookieV1(cookieDecoder, cookie);
-				}
+				return null;
 			}
-			return null;
+
+			if (cookies.ContainsKey(PxConstants.COOKIE_V3_PREFIX))
+			{
+				return new PxCookieV3(cookieDecoder, cookies[PxConstants.COOKIE_V3_PREFIX]);
+			}
+			
+			return new PxCookieV1(cookieDecoder, cookies[PxConstants.COOKIE_V1_PREFIX]);
```

Comment:
So now we simply assume that if prefix is not `COOKIE_V3_PREFIX` then it must be `PxCookieV1`?
I think that it is too coupled with the logic inside `PxContext`. In my view we should return `PxCookieV1` only if `cookies.ContainsKey(PxConstants.COOKIE_V1_PREFIX)` and otherwise return null (like we did before).

Reply:
we return null only if there's no cookie
if we find a cookie is either v3 or v1
actually I'd rather put v3 as default